rcl/src/rcl/arguments.c

```diff
@@ -134,6 +133,54 @@ _rcl_parse_log_level_rule(
   rcl_allocator_t allocator,
   int * log_level);
 
+/// Parse an argument that may or may not be a log file rule.
+/**
+ * \param[in] arg the argument to parse
+ * \param[in] allocator an allocator to use
+ * \param[in,out] log_config_file parsed log configuration file
+ * \return RCL_RET_OK if a valid log config file was parsed, or
+ * \return RCL_RET_BAD_ALLOC if an allocation failed, or
+ * \return RLC_RET_ERROR if an unspecified error occurred.
+ */
+RCL_LOCAL
+rcl_ret_t
+_rcl_parse_external_log_config_file(
+  const char * arg,
+  rcl_allocator_t allocator,
+  char ** log_config_file);
+
+/// Parse a bool argument that may or may not be for the provided key rule.
+/**
+ * \param[in] arg the argument to parse
+ * \param[in] key the key for the argument to parse. Should be a null terminated string
+ * \param[in] allocator an allocator to use
+ * \param[in,out] value parsed boolean value
+ * \return RCL_RET_OK if a valid log level was parsed, or
+ * \return RLC_RET_ERROR if an unspecified error occurred.
+ */
+RCL_LOCAL
+rcl_ret_t
+_rcl_parse_bool_arg(
+  const char * arg,
+  const char * key,
+  bool * value);
+
+/// Parse a null terminated string to a boolean value
+/**
+ *  The case sensitive values "T", "t", "True", "true", "Y", "y", "Yes", "yes", and "1" will all map to true
+ *  The case sensitive values "F", "f", "False", "false", "N", "n", "No", "no", and "0" will all map to false
+ *  \param[in] str a null terminated string to be parsed into a boolean
+ *  \param[in,out] val the boolean value parsed from the string. Left unchanged if string cannot be parsed to a valid bool
```

Comment:
nitpick: This line looks like it needs to be wrapped (our line length, even for docs, is 100). You can have the linter do it for you, with `ament_uncrustify . --reformat` in the directory of your project, in a shell with `install/setup.bash` sourced.